### PR TITLE
circuits: mpc-circuits: Refactor MPC circuits to operate over the Stark curve

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR refactors the implementation of the MPC circuits to work over the Stark curve. This is mostly changing interfaces to accommodate the new `async` flow of the MPC execution

### Testing
- Unit and integration tests pass, though CI will remain red throughout the refactor

### CODE REVIEW

File: circuits/integration/main.rs [hunk 1]
```diff
@@ -5,25 +5,16 @@
 mod mpc_circuits;
 mod mpc_gadgets;
 mod types;
-mod zk_circuits;
-mod zk_gadgets;
-
-use std::cell::Ref;
+// mod zk_circuits;
+// mod zk_gadgets;
```
Comment (joeykraut) [inline circuits/integration/main.rs:9]: I will update the ZK gadgets + circuits in a separate PR

File: circuits/integration/mpc_gadgets/bits.rs [hunk 2]
```diff
@@ -110,80 +99,51 @@ fn test_bit_add(test_args: &IntegrationTestArgs) -> Result<(), String> {
...
+
+    assert_scalar_batch_eq(&shared_bits_open, &vec![Scalar::zero(); shared_bits.len()])
 }
 
 /// Tests the to_bits_le gadget
 fn test_to_bits_le(test_args: &IntegrationTestArgs) -> Result<(), String> {
+    let fabric = &test_args.mpc_fabric;
     let value = 119;
 
     // The parties share the value 10 with little endian byte representation 0b0101
```
Comment (akirillo) [inline circuits/integration/mpc_gadgets/bits.rs:139]: nit: either remove this comment or update it:

```suggestion
    // The parties share the value 119 with little endian byte representation 0b11101110
```

File: test-helpers/src/mpc_network/mocks.rs [hunk 3]
```diff
@@ -79,47 +128,20 @@ impl MpcNetwork for MockNetwork {
...
-            .map_err(|e| MpcNetworkError::SerializationError(e.to_string()))?;
+        let msg = self.mock_conn.recv().await;
         Ok(msg)
     }
 
     async fn exchange_messages(
         &mut self,
         message: NetworkOutbound,
     ) -> Result<NetworkOutbound, MpcNetworkError> {
+        panic!("exchange called");
```
Comment (akirillo) [inline test-helpers/src/mpc_network/mocks.rs:144]: might be clearer to use an `unimplemented!()` macro here & remove the code below if this function is never meant to be called

**Review by akirillo — APPROVED** [review]
couple nits & couple missing test cases (I think) but correctness-wise lgtm

File: test-helpers/src/mpc_network/mocks.rs (hunk 3 above)
Comment (joeykraut) [reply test-helpers/src/mpc_network/mocks.rs:144]: Yeah oops, left that in for testing, the implementation here should be available, nice catch

**Pull request merged**